backends/p4tools/modules/testgen/CMakeLists.txt

```diff
@@ -114,7 +114,9 @@ target_link_libraries(
 add_custom_target(
   linkp4testgen
   # Add some convenience links for invoking p4testgen.
+  # TODO: How to best handle includes?
   COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_BINARY_DIR}/p4testgen ${CMAKE_BINARY_DIR}/p4testgen
```

Comment:
This is not related to the original issue, but I've just noticed you are linking `p4testgen` to top-level project, even in `p4c` is embedded into a downstream project using it.

I would suggest symlinking `p4testgen` into `${P4C_BINARY_DIR}` instead is a better choice (even though it can break some downstream workflow). This will leave the downstream projects to do the symlinking on their level if they so choose.

This is just a thought for future cleanup.